packages/camera/camera_avfoundation/ios/Classes/FLTCamMediaSettings.m

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTCamMediaSettings.h"
+
+@implementation FLTCamMediaSettings
+
+NS_INLINE void AssertPositiveNumberOrNil(NSNumber *_Nullable param,
+                                         const char *_Nonnull paramName) {
+  if (param != nil) {
+    NSCAssert(!isnan([param doubleValue]), @"%s is NaN", paramName);
+    NSCAssert([param doubleValue] > 0, @"%s is not positive: %@", paramName, param);
+  }
+}
+
+- (instancetype)initWithFramesPerSecond:(nullable NSNumber *)framesPerSecond
+                           videoBitrate:(nullable NSNumber *)videoBitrate
+                           audioBitrate:(nullable NSNumber *)audioBitrate
+                            enableAudio:(BOOL)enableAudio {
+  AssertPositiveNumberOrNil(framesPerSecond, "framesPerSecond");
+  AssertPositiveNumberOrNil(videoBitrate, "videoBitrate");
+  AssertPositiveNumberOrNil(audioBitrate, "audioBitrate");
+
```

Comment:
This needs to follow the standard pattern of an Obj-C initializer, calling `super init`. See other init methods in this plugin for standard format.

Reply:
super init pattern applied